scipy/optimize/_qap.py

```diff
@@ -0,0 +1,786 @@
+    >>> res = quadratic_assignment(A, B, method="2opt", options=options)
+    >>> print(res.fun)
+    46.47160735721583 # may vary
+
+    References
+    ----------
+    .. [1] J.T. Vogelstein, J.M. Conroy, V. Lyzinski, L.J. Podrazik,
+           S.G. Kratzer, E.T. Harley, D.E. Fishkind, R.J. Vogelstein, and
+           C.E. Priebe, "Fast approximate quadratic programming for graph
+           matching," PLOS one, vol. 10, no. 4, p. e0121002, 2015,
+           https://doi.org/10.1371/journal.pone.0121002
+
+    .. [2] D. Fishkind, S. Adali, H. Patsolic, L. Meng, D. Singh, V. Lyzinski,
+           C. Priebe, "Seeded graph matching", Pattern Recognit. 87 (2019):
+           203-215, https://doi.org/10.1016/j.patcog.2018.09.014
+    """
+
+    _check_unknown_options(unknown_options)
+
+    maxiter = operator.index(maxiter)
+
+    # ValueError check
+    A, B, partial_match = _common_input_validation(A, B, partial_match)
+
+    msg = None
+    if (A < 0).any() or (B < 0).any():
+        msg = "`A` and `B` matrices must contain only non-negative elements."
+    elif isinstance(init_J, str) and init_J not in {'barycenter'}:
+        msg = "Invalid 'init_J' parameter string"
+    elif init_weight is not None and (init_weight < 0 or init_weight > 1):
+        msg = "'init_weight' must be strictly between zero and one"
+    elif maxiter <= 0:
+        msg = "'maxiter' must be a positive integer"
+    elif tol <= 0:
+        msg = "'tol' must be a positive float"
+    if msg is not None:
+        raise ValueError(msg)
+
+    rng = check_random_state(rng)
+    n = A.shape[0]  # number of vertices in graphs
+    n_seeds = partial_match.shape[0]  # number of seeds
+    n_unseed = n - n_seeds
+
+    obj_func_scalar = 1
+    if maximize:
+        obj_func_scalar = -1
+
+    seed_dist_c = np.setdiff1d(range(n), partial_match[:, 1])
+    if shuffle_input:
+        seed_dist_c = rng.permutation(seed_dist_c)
+        # shuffle_input to avoid results from inputs that were already matched
+
+    seed_cost_c = np.setdiff1d(range(n), partial_match[:, 0])
+    perm_A = np.concatenate([partial_match[:, 0],
+                             seed_cost_c], axis=None).astype(int)
+    perm_B = np.concatenate([partial_match[:, 1],
+                             seed_dist_c], axis=None).astype(int)
+
+    A = A[np.ix_(perm_A, perm_A)]
+    B = B[np.ix_(perm_B, perm_B)]
+
+    # definitions according to Seeded Graph Matching [2].
+    A11 = A[:n_seeds, :n_seeds]
+    A12 = A[:n_seeds, n_seeds:]
+    A21 = A[n_seeds:, :n_seeds]
+    A22 = A[n_seeds:, n_seeds:]
+    B11 = B[:n_seeds, :n_seeds]
+    B12 = B[:n_seeds, n_seeds:]
+    B21 = B[n_seeds:, :n_seeds]
+    B22 = B[n_seeds:, n_seeds:]
+
+    # setting initialization matrix
+    if isinstance(init_J, str) and init_J == 'barycenter':
+        J = np.ones((n_unseed, n_unseed)) / float(n_unseed)
+    else:
+        _check_init_input(init_J, n_unseed)
+        J = init_J
+
+    if init_weight is not None:
```

Comment:
The docstring for `init_weight` specifies a float in the range [0, 1]. A value of `None` should raise a `ValueError`.

Reply:
For a fast path you can test for `init_weight=1`

Reply:
`init_weight` used to be `None` by default. I changed it when I removed `init_k` and this didn't get updated.